extensions/amp-story-auto-ads/0.1/story-ad-page.js

```diff
@@ -0,0 +1,486 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AnalyticsEvents,
+  AnalyticsVars,
+  STORY_AD_ANALYTICS,
+} from './story-ad-analytics';
+import {CommonSignals} from '../../../src/common-signals';
+import {CtaTypes} from './story-ad-localization';
+import {assertConfig} from '../../amp-ad-exit/0.1/config';
+import {assertHttpsUrl} from '../../../src/url';
+import {CSS as attributionCSS} from '../../../build/amp-story-auto-ads-attribution-0.1.css';
+import {
+  createElementWithAttributes,
+  elementByTag,
+  isJsonScriptTag,
+  iterateCursor,
+  openWindowDialog,
+  toggleAttribute,
+} from '../../../src/dom';
+import {createShadowRootWithStyle} from '../../amp-story/1.0/utils';
+import {dev, user, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getA4AMetaTags, getFrameDoc} from './utils';
+import {getServicePromiseForDoc} from '../../../src/service';
+import {parseJson} from '../../../src/json';
+import {setStyles} from '../../../src/style';
+
+/** @const {string} */
+const TAG = 'amp-story-auto-ads:page';
+
+/** @const {number} */
+const TIMEOUT_LIMIT = 10000; // 10 seconds
+
+/** @const {string} */
+const GLASS_PANE_CLASS = 'i-amphtml-glass-pane';
+
+/** @enum {string} */
+const PageAttributes = {
+  LOADING: 'i-amphtml-loading',
+  IFRAME_BODY_VISIBLE: 'amp-story-visible',
+};
+
+/** @enum {string} */
+const DataAttrs = {
+  CTA_TYPE: 'data-vars-ctatype',
+  CTA_URL: 'data-vars-ctaurl',
+};
+
+/** @enum {string} */
+const A4AVarNames = {
+  ATTRIBUTION_ICON: 'attribution-icon',
+  ATTRIBUTION_URL: 'attribution-url',
+  CTA_TYPE: 'cta-type',
+  CTA_URL: 'cta-url',
+};
+
+export class StoryAdPage {
+  /**
+   * @param {!Element} storyAutoAdsEl
+   * @param {!JsonObject} config
+   * @param {number} index
+   * @param {!./story-ad-localization.StoryAdLocalization} localization
+   */
+  constructor(storyAutoAdsEl, config, index, localization) {
```

Comment:
do you need the element? seems you only need `ampdoc`

Reply:
Changed to ampdoc